src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs

```diff
@@ -427,11 +433,12 @@ private async Task AcceptToMemoryPoolWorkerAsync(MempoolValidationState state, T
             // use the sequential scheduler for that.
             await this.mempoolLock.WriteAsync(() =>
             {
-                // is it already in the memory pool?
+                // Check if the transaction already exist in the mempool.
                 if (this.memPool.Exists(context.TransactionHash))
                 {
-                    this.logger.LogTrace("(-)[INVALID_ALREADY_EXISTS]");
-                    state.Invalid(MempoolErrors.InPool).Throw();
+                    state.Invalid(MempoolErrors.InPool);
```

Comment:
Perhaps add a comment explaining why we do this differently here to all the other checks. Also, `exists` in the current comment.